commands/new_function.go

```diff
@@ -81,12 +85,27 @@ the "Dockerfile" lang type in your YAML file.
 		return fmt.Errorf("you must supply a function language with the --lang flag")
 	}
 
-	PullTemplates("")
+	existingYaml := len(yamlFile) > 0
+	var stackFileName string
+	if !existingYaml {
+		// We will create a new YAML file for this function
+		stackFileName = functionName + ".yml"
+	} else {
+		stackFileName = yamlFile
+		// YAML file was passed in, so parse to see if it is valid
+		parsedServices, err := stack.ParseYAMLFile(stackFileName, "", "")
+		if err != nil || parsedServices == nil {
+			return fmt.Errorf("Specified file (" + stackFileName + ") is not valid YAML\n")
+		}
+		//TODO: Verify that services is good
+	}
 
 	if stack.IsValidTemplate(lang) == false {
 		return fmt.Errorf("%s is unavailable or not supported", lang)
 	}
 
+	PullTemplates("")
+
 	if _, err := os.Stat(functionName); err == nil {
 		return fmt.Errorf("folder: %s already exists", functionName)
```

Comment:
No checks are in place to ensure you don't end up with multiple identical function keys in the yaml, for example from repeated `new` commands where the function dir is deleted in between runs.
```
provider:
  name: faas
  gateway: http://localhost:8080
functions:
  newpython:
    lang: python
    handler: ./newpython
    image: newpython

  newpython2:
    lang: python
    handler: ./newpython2
    image: newpython2

  newpython2:
    lang: python
    handler: ./newpython2
    image: newpython2

  newpython2:
    lang: python
    handler: ./newpython2
    image: newpython2
```